interfaces/Referral-App/src/app/components/offer/offer.component.ts

```diff
@@ -16,5 +17,12 @@ export class OfferComponent {
   @Input()
   goBack;
 
-  constructor() {}
+  constructor(private loggingService: LoggingService) {}
+
+  public logClick(name) {
+    this.loggingService.logEvent('referral-offer-click', {
```

Comment:
I think we should put these event-names into a centrally stored Enum, so that we can see how many different ones there are and that we prevent typo's and get nice auto-complete in our editors.